network/tcp_client_test.cc

```diff
@@ -0,0 +1,281 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "tcp_client.h"
+#include "unix_domain_server.h"
+
+#include <gtest/gtest.h>
+#include <filesystem>
+#include <fstream>
+
+namespace Network
+{
+namespace
+{
+
+const std::string kTestHost = "127.0.0.1";
```

Comment:
Static variables with non-trivial destructors (like `std::string`) can be dangerous since they may still be used after they are destroyed. See https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables for more info. Here's some more information from Abseil: https://abseil.io/tips/140#approaches-that-work-in-source-files

Here, a better option would be `std::string_view`.  Also, consider `constexpr`.

```suggestion
constexpr std::string_view kTestHost = "127.0.0.1";
```